src/main/java/com/minecolonies/client/gui/WindowHutGuardTower.java

```diff
@@ -0,0 +1,476 @@
+        }
+
+        if(patrolManually && task.equals(BuildingGuardTower.Task.PATROL))
+        {
+            givePlayerScepter(BuildingGuardTower.Task.PATROL);
+            LanguageHandler.sendPlayerLocalizedMessage(player, "com.minecolonies.job.guard.tool.taskPatrol");
+        }
+        else if(task.equals(BuildingGuardTower.Task.GUARD))
+        {
+            givePlayerScepter(BuildingGuardTower.Task.GUARD);
+            LanguageHandler.sendPlayerLocalizedMessage(player, "com.minecolonies.job.guard.tool.taskGuard");
+        }
+        window.close();
+    }
+
+
+    /**
+     * Send message to player to add scepter to his inventory.
+     * @param localTask the task to execute with the scepter.
+     */
+    private void givePlayerScepter(BuildingGuardTower.Task localTask)
+    {
+        MineColonies.getNetwork().sendToServer(new GuardScepterMessage(localTask.ordinal(), building.getID()));
+    }
+
+    /**
+     * Sends changes to the server.
+     */
+    private void sendChangesToServer()
+    {
+        int ordinal = building.job == null ? -1 : job.ordinal();
+        MineColonies.getNetwork().sendToServer(new GuardTaskMessage(building, ordinal, assignManually, patrolManually, retrieveOnLowHealth, task.ordinal()));
+    }
+
+    /**
+     * Switch the retrieval mode.
+     * @param button clicked button
+     */
+    private void switchRetrievalMode(final Button button)
+    {
+        building.retrieveOnLowHealth = !building.retrieveOnLowHealth;
+        pullInfoFromHut();
+        sendChangesToServer();
+        this.findPaneOfTypeByID(BUTTON_RETRIEVAL_MODE, Button.class).setLabel(retrieveOnLowHealth ? ON : OFF);
+    }
+
+    /**
+     * Switch the patrol mode.
+     * @param button clicked button
+     */
+    private void switchPatrolMode(final Button button)
+    {
+        building.patrolManually = !building.patrolManually;
+        pullInfoFromHut();
+        sendChangesToServer();
+        handleButtons();
+    }
+
+    /**
+     * Switch the job.
+     * @param button clicked button
+     */
+    private void switchJob(final Button button)
+    {
+        if(building.job == null)
+        {
+            building.job = BuildingGuardTower.GuardJob.RANGER;
+        }
+        else
+        {
+            if (building.job.equals(BuildingGuardTower.GuardJob.KNIGHT))
+            {
+                building.job = BuildingGuardTower.GuardJob.RANGER;
+            }
+            else
+            {
+                building.job = BuildingGuardTower.GuardJob.KNIGHT;
+            }
+        }
+        pullInfoFromHut();
+        sendChangesToServer();
+    }
+
+    /**
+     * Switch the assignment mode.
+     * @param button clicked button
+     */
+    private void switchAssignmentMode(final Button button)
+    {
+        building.assignManually = !building.assignManually;
+        pullInfoFromHut();
+        sendChangesToServer();
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)